### ISSUE TITLE
Same member groups in multibyte characters are separately generated.

### ISSUE THREAD

**Issue by st-masa** [doxygen/doxygen#9566]
**Describe the bug**

When I configure the same member group names in multibyte characters, these member groups are separately generated in the output html.

**Expected behavior**

The member groups with the same name should be integrated even though written in multibyte characters.

**Screenshots**
![MemberGroup_separated](https://user-images.githubusercontent.com/79039413/188252705-cf41d12c-2be1-4d61-97ea-ed4f6ab2323f.png)

**To Reproduce**

When using @name commands in ascii, member groups are integrated correctly.
However, @name commands in multibyte characters cause separation.

The attachment [MemberGroupSample.zip](https://github.com/doxygen/doxygen/files/9481666/MemberGroupSample.zip) contains two header files that contain the same member groups.

```
/**
    @file
    @brief  header1
 */

namespace common {

// (example of success)
//! @name Member group
//! @{

/**
    @brief    Function1_1

    @param[in] arg1 This is arg1
    @param[in] arg2 This is arg2
 */
void Function1_1(int arg1, int arg2);

//! @}

// (example of failure)
//! @name  成员分组
//! @{

/**
    @brief    Function1_2

    @param[in] arg1 This is arg1
    @param[in] arg2 This is arg2
 */
void Function1_2(int arg1, int arg2);

//! @}

// (example of failure)
//! @name  メンバーグループ
//! @{

/**
    @brief    Function1_3

    @param[in] arg1 This is arg1
    @param[in] arg2 This is arg2
 */
void Function1_3(int arg1, int arg2);

//! @}

}
```

**Version**
1.9.5 and 1.9.4 on Windows 10 (64 bit) and MacOSX

**Additional context**

The change of `qstricmp` function in [Refactoring: replace old style casts](https://github.com/doxygen/doxygen/commit/446038e8e475bfd17836163d26e0d58c40a5fdbb) may cause this issue.
While changed `qstrnicmp` may not cause the issue.

**Comment by st-masa** [comment]
Although I haven't checked,  this bug may affect some places where `qstricmp` is used on dealing with multibyte inputs.
https://github.com/doxygen/doxygen/search?p=1&q=qstricmp

**Comment by albert-github** [comment]
The mentioned commit indeed looks like to be the problem as before it the output is OK and with this refactoring it isn't.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9574

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
